crates/hir_ty/src/tests/traits.rs

```diff
@@ -1936,14 +1936,14 @@ fn closure_2() {
             84..85 'f': F
             84..88 'f(1)': {unknown}
             86..87 '1': i32
-            98..99 'g': |u64| -> i32
-            102..111 '|v| v + 1': |u64| -> i32
+            98..99 'g': |u64| -> {unknown}
+            102..111 '|v| v + 1': |u64| -> {unknown}
             103..104 'v': u64
             106..107 'v': u64
-            106..111 'v + 1': i32
+            106..111 'v + 1': {unknown}
             110..111 '1': i32
-            117..118 'g': |u64| -> i32
-            117..124 'g(1u64)': i32
+            117..118 'g': |u64| -> {unknown}
+            117..124 'g(1u64)': {unknown}
```

Comment:
This involves a general `InferenceVar` which results in these not resolving anymore since we can't make assumptions for those

Reply:
Hm. Can we add the necessary impls to the test to make this work?

Reply:
Oops, ye should've done that from the start. Resolves with those impls added.